src/Bicep.RoslynAnalyzers/LinterRuleTypeGenerator.cs

```diff
@@ -0,0 +1,196 @@
+                return null;
+            }
+
+            var classSyntax = method.FirstAncestorOrSelf<ClassDeclarationSyntax>();
+            if (classSyntax is null)
+            {
+                return null;
+            }
+
+            if (syntaxContext.SemanticModel.GetDeclaredSymbol(classSyntax, cancellationToken: cancellationToken) is not INamedTypeSymbol classSymbol)
+            {
+                return null;
+            }
+
+            return new(method, classSymbol);
+        }
+
+        private static LinterRuleClassContext? TryGetLinterRuleContext(GeneratorSyntaxContext syntaxContext, CancellationToken cancellationToken)
+        {
+            if (syntaxContext.Node is not ClassDeclarationSyntax @class ||
+                syntaxContext.SemanticModel.GetDeclaredSymbol(@class) is not INamedTypeSymbol classSymbol)
+            {
+                // shouldn't happen
+                return null;
+            }
+
+            if(classSymbol.IsAbstract)
+            {
+                // reject abstract classes
+                return null;
+            }
+
+            var ruleTypeSymbol = syntaxContext.SemanticModel.Compilation.GetTypeByMetadataName("Bicep.Core.Analyzers.Interfaces.IBicepAnalyzerRule");
+            if(ruleTypeSymbol is null)
+            {
+                // bicep rule interface doesn't exist
+                return null;
+            }
+
+            if(!syntaxContext.SemanticModel.Compilation.ClassifyConversion(classSymbol, ruleTypeSymbol).IsImplicit)
+            {
+                // class does not implement the rule interface
+                return null;
+            }
+
+            return new(classSymbol);
+        }
+
+        private static bool IsGeneratorAttribute(GeneratorSyntaxContext syntaxContext, AttributeSyntax attributeSyntax, CancellationToken cancellationToken)
+        {
+            // get attribute constructor symbol
+            if (syntaxContext.SemanticModel.GetSymbolInfo(attributeSyntax, cancellationToken).Symbol is not IMethodSymbol attributeConstructorSymbol)
+            {
+                return false;
+            }
+
+            // attribute constructor symbol should be contained in an attribute type symbol
+            var attributeTypeName = attributeConstructorSymbol.ContainingType?.ToDisplayString();
+
+            // c# type names are case-sensitive
+            return string.Equals(attributeTypeName, FullyQualifiedAttributeName, StringComparison.Ordinal);
+        }
+
+        private void GenerateSource(SourceProductionContext context, (GeneratorMethodContext? generatorContext, ImmutableArray<LinterRuleClassContext?> linterRules) info)
+        {
+            var (generatorContext, linterRules) = info;
+
+            if (generatorContext?.ClassSymbol.ContainingNamespace is null)
+            {
+                return;
+            }
+
+            var builder = new StringBuilder();
+            builder.AppendLine(@"// Copyright (c) Microsoft Corporation.
```

Comment:
>builder [](http://example.com/codeflow?start=12&length=7)

T4 doesn't work in .net core. There is a ported T4 engine available but it doesn't have good design-time support in VS, so decided not to use it.

Reply:
I'm also having issues with `dotnet publish` and a `ProjectReference`-referenced generator, so I'm going to put the generator in the shared tools repo. This will also prevent the problem you mentioned from happening.